crates/ruff_workspace/src/options.rs

```diff
@@ -2762,10 +2760,18 @@ pub struct PydocstyleOptions {
 
 impl PydocstyleOptions {
     pub fn into_settings(self) -> pydocstyle::settings::Settings {
-        pydocstyle::settings::Settings {
-            convention: self.convention,
-            ignore_decorators: BTreeSet::from_iter(self.ignore_decorators.unwrap_or_default()),
-            property_decorators: BTreeSet::from_iter(self.property_decorators.unwrap_or_default()),
+        let PydocstyleOptions {
+            convention,
+            ignore_decorators,
+            property_decorators,
+        } = self;
+        let settings = pydocstyle::settings::Settings::default()
+            .with_ignore_decorators(ignore_decorators.unwrap_or_default())
+            .with_property_decorators(property_decorators.unwrap_or_default());
+        if let Some(convention) = convention {
+            settings.with_convention(convention)
+        } else {
+            settings
         }
     }
```

Comment:
I personally prefer the old code here because it guarantees that all properties get initialized. The fact that it leaks the `BTreeSet` internals is unfortunate but I rather have compile time guarantees (and guidance) on how new settings have to be added.

Reply:
Yeah. I wondered about just creating a `Settings::new()` method, but went for this as I thought `Settings::default().with_property_decorators(&[...])` was more readable than `Settings::new(None, &[...], &[])`. It's hard to tell with the `new()` call which decorator list is empty and which isn't, and having to pass in `None` explicitly for the `convention` argument feels awkward.

But it's less code and it guarantees that all properties get initialized, so I guess I'll go for that instead.